pilot/pkg/model/config_test.go

```diff
@@ -177,42 +178,70 @@ func TestServiceKey(t *testing.T) {
 func TestSubsetKey(t *testing.T) {
 	hostname := model.Hostname("hostname")
 	cases := []struct {
+		got      string
 		hostname model.Hostname
 		subset   string
 		port     *model.Port
 		want     string
+		err      error
 	}{
 		{
 			hostname: "hostname",
 			subset:   "subset",
 			port:     &model.Port{Name: "http", Port: 80, Protocol: model.ProtocolHTTP},
-			want:     "outbound|http|subset|hostname",
+			want:     "outbound.http.subset.hostname",
 		},
 		{
 			hostname: "hostname",
 			subset:   "subset",
 			port:     &model.Port{Port: 80, Protocol: model.ProtocolHTTP},
-			want:     "outbound||subset|hostname",
+			want:     "outbound.80.subset.hostname",
 		},
 		{
 			hostname: "hostname",
 			subset:   "",
 			port:     &model.Port{Name: "http", Port: 80, Protocol: model.ProtocolHTTP},
-			want:     "outbound|http||hostname",
+			want:     "outbound.http._default_.hostname",
+		},
+		{
+			want:   "badClusterKey",
+			subset: "",
+			port:   &model.Port{Name: "dontcare", Port: 80, Protocol: model.ProtocolHTTP},
+			got:    "badClusterKey",
+			err:    fmt.Errorf("malformed cluster-key"),
 		},
 	}
 
 	for _, c := range cases {
-		got := model.BuildSubsetKey(model.TrafficDirectionOutbound, c.subset, hostname, c.port)
-		if got != c.want {
-			t.Errorf("Failed: got %q want %q", got, c.want)
-		}
+		t.Run(c.want, func(t *testing.T) {
+			got := c.got
+			if got == "" {
+				got = model.BuildSubsetKey(model.TrafficDirectionOutbound, c.subset, hostname, c.port)
```

Comment:
Thanks for doing the UTs, this is useful for documenting behavior as well. 
Couple of nits here:
 - rename "got" field in struct to "gotOverride" otherwise it's harder to understand what's going on. 
 - prefer to condition the special case i.e. 
got := model.BuildSubsetKey(...
if c.gotOverride != "" {
  got = c.gotOverride
}